src/main/java/edu/kis/vh/nursery/collection/IntArrayStack.java

```diff
@@ -0,0 +1,43 @@
+package edu.kis.vh.nursery.collection;
+
+public class IntArrayStack implements StackInterface {
+
+    private static final int NUMBERS_CAPACITY = 12;
+
+    private int[] numbers = new int[NUMBERS_CAPACITY];
+
+    private int total = EMPTY_STACK_VALUE;
+
+    @Override public void push(final int in) {
+        if (!isFull())
+            numbers[total++] = in;
+    }
+
+    @Override public boolean isEmpty() {
+        return total == EMPTY_STACK_VALUE;
+    }
+
+    @Override public boolean isFull() {
+        return total == NUMBERS_CAPACITY + EMPTY_STACK_VALUE;
```

Comment:
There is no reasonable relation between the default returned value and isFull logic!

Changing IS_EMPTY value requires such fixes? It does not make sense. Incorrect fix.